tests/test_packages/test_connections/test_http/test_http.py

```diff
@@ -0,0 +1,347 @@
+
+        async def client_thread():
+            executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+            loop = asyncio.get_event_loop()
+            result = await loop.run_in_executor(executor, request_response_cycle)
+            return result
+
+        response_code, response_body = await client_thread()
+
+        assert response_code == 404 and response_body == b"Request Not Found"
+
+    @pytest.mark.asyncio
+    async def test_get_408(self):
+        """Test get 408."""
+
+        def request_response_cycle():
+            conn = http.client.HTTPConnection(self.host, self.port)
+            conn.request("GET", "/pets")
+            response = conn.getresponse()
+            return response.status, response.read()
+
+        async def client_thread():
+            executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+            loop = asyncio.get_event_loop()
+            result = await loop.run_in_executor(executor, request_response_cycle)
+            return result
+
+        response_code, response_body = await client_thread()
+
+        assert response_code == 408 and response_body == b"Request Timeout"
+
+        envelope = await self.http_connection.receive()
+        assert (
+            envelope is not None
+            and envelope.to == self.address
+            and len(self.http_connection.channel.timed_out_request_ids) == 1
+        )
+
+    # @pytest.mark.asyncio
+    # async def test_get_202(self):
+    #     """Test send connection error."""
+
+    #     def request_response_cycle() -> Tuple[int, bytes]:
+    #         conn = http.client.HTTPConnection(self.host, self.port)
+    #         conn.request("GET", "/pets")
+    #         response = conn.getresponse()
+    #         return response.status, response.read()
+
+    #     async def client_thread() -> Tuple[int, bytes]:
+    #         executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+    #         loop = asyncio.get_event_loop()
+    #         result = await loop.run_in_executor(executor, request_response_cycle)
+    #         return result
+
+    #     response_code, response_body = await client_thread()
+
+    #     async def agent_processing() -> None:
+    #         # we block here to give it some time for the envelope to make it to the queue
+    #         await asyncio.sleep(1)
+    #         envelope = await self.http_connection.receive()
+    #         if envelope is not None:
+    #             response_envelope = Envelope(
+    #                 to=envelope.sender,
+    #                 sender=envelope.to,
+    #                 protocol_id=envelope.protocol_id,
+    #                 context=envelope.context,
+    #                 message=json.dumps({"status_code": 200, "message": "Response"}).encode(),
+    #             )
+    #         await self.http_connection.send(response_envelope)
+
+    #     agent_processing()
+
+    #     assert response_code == 200 and b"Response"
```

Comment:
this should have `response_body == b"Response"`

however, in the other branch, I didn't manage to make it return the bytes, I have to investigate more. Feel free to take the token.